domain/application/service/application.go

```diff
@@ -26,47 +26,60 @@ import (
 	"github.com/juju/juju/internal/storage"
 )
 
+// AtomicApplicationState describes retrieval and persistence methods for
+// applications that require atomic transactions.
+type AtomicApplicationState interface {
+	domain.AtomicStateBase
+
+	// GetApplicationID returns the ID for the named application, returning an
+	// error satisfying [applicationerrors.ApplicationNotFound] if the
+	// application is not found.
+	GetApplicationID(ctx domain.AtomicContext, name string) (coreapplication.ID, error)
+
+	// UpsertUnit creates or updates the specified application unit, returning
+	// an error satisfying [applicationerrors.ApplicationNotFoundError] if the
+	// application doesn't exist.
+	UpsertUnit(domain.AtomicContext, coreapplication.ID, application.UpsertUnitArg) error
+
+	// ApplicationScaleState looks up the scale state of the specified
+	// application, returning an error satisfying
+	// [applicationerrors.ApplicationNotFound] if the application is not found.
+	ApplicationScaleState(domain.AtomicContext, coreapplication.ID) (application.ScaleState, error)
+
+	// UnitLife looks up the life of the specified unit, returning an error
+	// satisfying [uniterrors.NotFound] if the unit is not found.
+	UnitLife(ctx domain.AtomicContext, unitName string) (life.Life, error)
+}
+
 // ApplicationState describes retrieval and persistence methods for
 // applications.
 type ApplicationState interface {
-	domain.AtomicStateBase
+	AtomicApplicationState
+
 	// StorageDefaults returns the default storage sources for a model.
 	StorageDefaults(ctx context.Context) (domainstorage.StorageDefaults, error)
 
-	// GetStoragePoolByName returns the storage pool with the specified name, returning an error
-	// satisfying [storageerrors.PoolNotFoundError] if it doesn't exist.
+	// GetStoragePoolByName returns the storage pool with the specified name,
```

Comment:
Don't like these changes to comments. The line length is still < the method signature length. We're not in the 1970s any more. They read fine to me. YMMV I guess.

Reply:
This was decided in the last sprint. This was because it's helpful for code reviews in the github UI, where split view makes it more difficult with longer lines. If we have a standard then we might as well be consistent. We settled on:

 - 80 width for comments (URLs and the new error constants are exceptions)
 - 120 width for code

P.S. you where there!